app/kubemci/pkg/kubeutils/utils.go

```diff
@@ -217,3 +222,53 @@ func ObjectMetaAndSpecEquivalent(a, b runtime.Object, ignoreAnnotationKeys map[s
 	specB := reflect.ValueOf(b).Elem().FieldByName("Spec").Interface()
 	return ObjectMetaEquivalent(objectMetaA, objectMetaB, ignoreAnnotationKeys) && reflect.DeepEqual(specA, specB)
 }
+
+// GetServiceNodePort takes an IngressBackend and returns a corresponding ServicePort
+func GetServiceNodePort(be v1beta1.IngressBackend, namespace string, client kubeclient.Interface) (ingressbe.ServicePort, error) {
+	svc, err := getSvc(be.ServiceName, namespace, client)
+	// Refactor this code to get serviceport from a given service and share it with kubernetes/ingress.
+	appProtocols, err := annotations.SvcAnnotations(svc.GetAnnotations()).ApplicationProtocols()
+	if err != nil {
+		return ingressbe.ServicePort{}, err
+	}
+
+	var port *v1.ServicePort
+PortLoop:
+	for _, p := range svc.Spec.Ports {
+		np := p
+		switch be.ServicePort.Type {
+		case intstr.Int:
+			if p.Port == be.ServicePort.IntVal {
+				port = &np
+				break PortLoop
+			}
+		default:
+			if p.Name == be.ServicePort.StrVal {
+				port = &np
+				break PortLoop
+			}
+		}
+	}
+
+	if port == nil {
+		return ingressbe.ServicePort{}, fmt.Errorf("could not find matching nodeport for backend %+v and service %s/%s. Looking for port %+v in %v", be, namespace, be.ServiceName, be.ServicePort, svc.Spec.Ports)
+	}
+
+	proto := ingressutils.ProtocolHTTP
+	if protoStr, exists := appProtocols[port.Name]; exists {
+		glog.V(2).Infof("service %s/%s: using protocol to %q", namespace, be.ServiceName, protoStr)
+		proto = ingressutils.AppProtocol(protoStr)
+	}
+
+	p := ingressbe.ServicePort{
+		Port:     int64(port.NodePort),
+		Protocol: proto,
+		SvcName:  types.NamespacedName{Namespace: namespace, Name: be.ServiceName},
+		SvcPort:  be.ServicePort,
+	}
+	glog.Infof("Found ServicePort: %+v", p)
+	return p, nil
+}
```

Comment:
Add a line break

Reply:
done. (go fmt enforces 1 newline between functions, it appears. I put the newline before the brace.)